segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -252,6 +260,29 @@ public void append(Append append) {
                 });
     }
 
+    @Override
+    public void createTransientSegment(CreateTransientSegment createTransientSegment) {
+        String operation = "createTransientSegment";
+        long traceId = LoggerHelpers.traceEnter(log, operation, createTransientSegment);
+        Collection<AttributeUpdate> attributes = Arrays.asList(
+                new AttributeUpdate(CREATION_TIME, AttributeUpdateType.None, System.currentTimeMillis()),
+                new AttributeUpdate(ATTRIBUTE_SEGMENT_TYPE, AttributeUpdateType.None, SegmentType.TRANSIENT_SEGMENT.getValue())
+        );
+        String transientSegmentName = NameUtils.getTransientNameFromId(createTransientSegment.getParentSegment(), UUID.randomUUID());
+        store.createStreamSegment(transientSegmentName, SegmentType.TRANSIENT_SEGMENT, attributes, TIMEOUT)
```

Comment:
This needs to be registered to be removed upon drop before the operations is invoked on the store.

Reply:
@tkaitchuck This work should be done now. We need to validate that upon a connection close, the `AppendProcessor.close` method is also invoked.